x/axelarnet/keeper/keeper.go

```diff
@@ -58,6 +60,23 @@ func (k Keeper) SetParams(ctx sdk.Context, n types.Nexus, p types.Params) {
 	}
 }
 
+// SetDepositAddress set the deposit address for the given recipient
+func (k Keeper) SetDepositAddress(ctx sdk.Context, recipient nexus.CrossChainAddress, address sdk.AccAddress) {
+	key := depositAddrPrefix.Append(utils.LowerCaseKey(recipient.String()))
+	k.getStore(ctx).SetRaw(key, address.Bytes())
+}
+
+// GetDepositAddress gets the deposit address for the given recipient
+func (k Keeper) GetDepositAddress(ctx sdk.Context, recipient nexus.CrossChainAddress) (sdk.AccAddress, bool) {
+	key := depositAddrPrefix.Append(utils.LowerCaseKey(recipient.String()))
+	bz := k.getStore(ctx).GetRaw(key)
+	if bz == nil {
+		return sdk.AccAddress{}, false
+	}
+
+	return sdk.AccAddress(bz), true
+}
+
```

Comment:
isn't this better implemented in the nexus module so we can reuse it everywhere?

Reply:
good point, but then we wouldn't be able to return a specific type. We would need to always provide and return a string.  But in any case, we probably should address this in a different PR.

Reply:
yes, it would be serialized, but in the general case that is more common to have anyway, right?